streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java

```diff
@@ -382,6 +383,143 @@
      */
     <KR> KStream<KR, V> toStream(final KeyValueMapper<? super K, ? super V, ? extends KR> mapper);
 
+    /**
+     * Transform the value of each input record into a new value (with possible new type) of the output record.
```

Comment:
nit for first sentence: `Create a new {@code KTable} by transforming the value of each record in this {@code KTable} into a new value.` Also we can add one more sentence comparing with `mapValues` that this API allows users to have more flexible custom transformation logic that can rely on additional state stores and record context, plus they can add period functions via scheduling etc.

Ditto below.

Reply:
docs already calls out the potential use of `punctuate`. I'll add something for more flexibility / state stores and record context.